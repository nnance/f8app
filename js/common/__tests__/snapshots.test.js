import React from 'react';
import renderer from 'react-test-renderer';

describe('CircleImageWithCategory', () => {
  const CircleImageWithCategory = require('../CircleImageWithCategory').default;
  it('render CircleImageWithCategory', () => {
    const tree = renderer.create(<CircleImageWithCategory
      source={{ uri: 'x.jpg' }}
      categorySource={{ uri: 'x.jpg' }}
      size={100}
    />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe('ModalSpinner', () => {
  const ModalSpinner = require('../ModalSpinner').default;
  it('render visible', () => {
    const tree = renderer.create(<ModalSpinner visible />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('render invisible', () => {
    const tree = renderer.create(<ModalSpinner visible={false} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
