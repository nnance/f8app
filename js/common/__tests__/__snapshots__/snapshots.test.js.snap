exports[`CircleImageWithCategory render CircleImageWithCategory 1`] = `
<View
  style={
    Object {
      "height": 100,
      "width": 100,
    }
  }>
  <View
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }>
    <View
      style={
        Object {
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "borderRadius": 50,
          "height": 100,
          "left": -0,
          "position": "absolute",
          "top": -0,
          "width": 100,
        }
      } />
    <Image
      categorySource={
        Object {
          "uri": "x.jpg",
        }
      }
      size={100}
      source={
        Object {
          "uri": "x.jpg",
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 50,
            "height": 100,
            "width": 100,
          },
          Array [
            undefined,
            Object {
              "backgroundColor": "transparent",
            },
          ],
        ]
      }
      subSource={1} />
    <View
      style={
        Object {
          "height": 100,
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": 100,
        }
      } />
  </View>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 3.125,
          "height": 25,
          "position": "absolute",
          "right": 3.125,
          "width": 25,
        },
        undefined,
      ]
    } />
</View>
`;

exports[`ModalSpinner render invisible 1`] = `
<Modal
  transparent={true}
  visible={false}>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 255, 255, 0.6)",
        "flex": 1,
        "justifyContent": "center",
      }
    }>
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="large"
      style={
        Array [
          Object {
            "height": 80,
          },
        ]
      } />
  </View>
</Modal>
`;

exports[`ModalSpinner render visible 1`] = `
<Modal
  transparent={true}
  visible={true}>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 255, 255, 0.6)",
        "flex": 1,
        "justifyContent": "center",
      }
    }>
    <ActivityIndicator
      animating={true}
      color="#999999"
      hidesWhenStopped={true}
      size="large"
      style={
        Array [
          Object {
            "height": 80,
          },
        ]
      } />
  </View>
</Modal>
`;
